### PR TITLE
Add RemoveDCOffset effect

### PR DESCRIPTION
**Pull request by PatrickKallenbach**
Resolves: https://github.com/audacity/audacity/issues/2408

The original issue was requesting a DC Offset tool similar to what the Normalize effect can do before normalization. In my university class, we were assigned to find and contribute to an open source project, so I thought I should contribute to this project which I have used myself in the past. The new effect is modified from the existing Normalize effect to automatically remove the offset, bypassing the menu similar to the Invert effect. I am certain improvements could be made upon the modified files, perhaps adding more features, or removing steps that were required for the Normalize effect but not for DC offset removal. I also only added the effects to the src/effects folder, but I do not know if there is something else that needs to know of the updated effects, since it does not currently categorize with the other related effects.

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

**Comment by Paul-Licameli** [comment]
This is not a complete code review, but in fact -- the .h file is not needed at all.   You can just define the class locally in the .cpp file.  More than that -- I would also wrap the whole thing in an anonymous namespace, and save a little bit of space in the symbol tables that the linker has to cope with.

Audacity wasn't originally written this way back in the late twentieth century, but nowadays, static registration objects do a lot of magic to push things into registries (of effects, macro commands, whatever) allowing scattered definitions of details, and making it unnecessary for those details to expose a header file.

**Comment by LWinterberg** [comment]
> I do not know if there is something else that needs to know of the updated effects, since it does not currently categorize with the other related effects.

That's in resources/EffectsMenuDefaults.xml, probably in the "noise removal and repair" category.

File: RemoveDCOffset.cpp [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+            msg = topMsg +
+               // TODO: more-than-two-channels-message
+               XO("Processing first track of stereo pair: %s")
+                  .Format(trackName);
+
+         // Use multiplier in the second, processing loop over channels
+         auto pOffset = offsets.begin();
+         auto pExtent = extents.begin();
+         for (const auto channel : channels) {
+            const auto extent = oneChannel ? *pExtent++: maxExtent;
```
Comment (saintmatthieu) [inline RemoveDCOffset.cpp:183]: Hey Patrick, thanks for your contribution!
It looks like you don't use `extent`, in the end.

**Comment by Paul-Licameli** [comment]
Anonymous namespace just means this -- add the line
`namespace {`
after any #include directives and forward declarations of incomplete types, then at the bottom of the file,
'}'

**Comment by petersampsonaudacity** [comment]
@LWinterberg   

If/when this PR gets pulled for this new effect, are you planniing to remove the DC offset removal from the Normalize effect or just leave it in there?

**Comment by LWinterberg** [comment]
Hello! We just [switched master to 4.0.0](https://forum.audacityteam.org/t/master-has-been-switched-to-4-0-0-development/120444) which makes this PR incompatible and/or obsolete as we're switching UI framework and reorganizing large parts of the codebase. You're very welcome to make these kinds of contributions in the future though, after things stabilize a little bit.

Also, sorry for this comically large delay.

**Pull request closed without merging**